nbextensions/appCell2/widgets/tabs/resultsViewer.js

```diff
@@ -90,17 +101,36 @@ define([
             // the job output
             if (!reportInputs) {
                 return Promise.try(() => {
+                    const viewerName = result ? result.name : null;
                     const jobOutput = jobState.job_output
                         ? jobState.job_output.result
                         : 'no output found';
+                    if (viewerName === 'text-only') {
+                        ui.setContent('results.body', buildOutputText(result.params.result_text));
+                    } else {
+                        ui.setContent('results.body', ui.buildPresentableJson(jobOutput));
+                    }
                     ui.getElement('results').classList.remove('hidden');
-                    ui.setContent('results.body', ui.buildPresentableJson(jobOutput));
                 });
             }
             // otherwise, render the report
             return renderReportView(reportInputs);
         }
 
+        function buildOutputText(resultText) {
+            // default if text is empty, null, or undefined
+            if (resultText === null || resultText === undefined || resultText === '') {
```

Comment:
Did you want to check for arrays / maps here as well?

Reply:
Urp, yeah, I meant to add that.